service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -194,7 +194,8 @@ func (policy *selectedOrAllAPIsForwardingRedirectionPolicy) WithDomainIDRedirect
 	if err != nil {
 		return err
 	}
-	if domainEntry.IsDeprecatedOrDeleted() {
+	// Skip domain deprecation check for ListWorkflowExecutions API
+	if apiName != "ListWorkflowExecutions" && domainEntry.IsDeprecatedOrDeleted() {
```

Comment:
Instead having a hardcoded endpoint here in this function let's define a new category of APIs that are allowed for deprecated domains. So maybe `allowedAPIsForDeprecatedDomains map[string]struct{}` at the top of this file similar to those other maps there